File: LeetCode_Algorithm/409/409.js
var longestPalindrome = function(s) {

  // O(n) 145ms 5%
  let hashMap = {}
  for (let ch of s) {
    if (!hashMap[[ch]]) {
      hashMap[ch] = 1
    } else {
      hashMap[ch]++;
    }
  }
  let ans = 0;
  for (let key in hashMap) {
    ans += parseInt(hashMap[key] / 2) * 2
    if (ans % 2 == 0 && hashMap[key] % 2 == 1){
      ans += 1
    }
  }
  return ans
}

// console.log(longestPalindrome("aabba"))
console.log(longestPalindrome("abccccdd"))

File: LeetCode_Algorithm/409/409.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hashMap = {}
        for char in s:
            if char not in hashMap:
                hashMap[char] = 1
            else:
                hashMap[char] += 1

        ans = 0
        for key in hashMap:
            ans += (hashMap[key] // 2) * 2
            if ans % 2 == 0 and hashMap[key] % 2 == 1:
                ans += 1
        return ans


print(Solution().longestPalindrome("abccccdd"))